### PR TITLE
[VPU][GT] Add pass for marking fast stages

### PR DESCRIPTION
**Pull request by AndrewBakalinIntel**
Ticket - #-35904
It was discovered that it is sometimes useful to mark fast stages (e.g. stages that process less than 100 elements) in order to be able to parse them from final performance report and estimate its contribution into performance.

### CODE REVIEW

File: inference-engine/src/vpu/graph_transformer/src/middleend/passes/mark_fast_stages.cpp [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+        for (const auto& stage : model->getStages()) {
+            bool isStageFast = true;
+
+            for (const auto& output : stage->outputs()) {
+                if (output->desc().totalDimSize() > 100) {
+                    isStageFast = false;
+                    break;
+                }
+            }
```
Comment (gladilov-gleb) [inline inference-engine/src/vpu/graph_transformer/src/middleend/passes/mark_fast_stages.cpp:24]:
```cpp
const auto& outputs = stage->outputs();
const bool isSlow = std::find_if(outputs.cbegin(), outputs.cend(), [](const Data& output) { return output->desc().totalDimsSize() > 100; });
```

File: inference-engine/tests/unit/vpu/middleend_tests/passes_tests/mark_fast_stages.cpp [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    _testModel.createInputs({desc});
+    _testModel.createOutputs({desc});
+
+    auto stage = _testModel.addStage({InputInfo::fromNetwork()}, {OutputInfo::fromNetwork()});
+    ASSERT_NO_THROW(Compile());
+
+    ASSERT_EQ(stage->name().find("@fast-stage"), std::string::npos);
+}
+
+} // namespace vpu
```
Comment (gladilov-gleb) [inline inference-engine/tests/unit/vpu/middleend_tests/passes_tests/mark_fast_stages.cpp:62]: Nitpick: no empty line at eof?

File: inference-engine/src/vpu/graph_transformer/src/middleend/passes/mark_fast_stages.cpp (hunk 1 above)
Comment (AndrewBakalinIntel) [reply inference-engine/src/vpu/graph_transformer/src/middleend/passes/mark_fast_stages.cpp]: Fixed.

File: inference-engine/tests/unit/vpu/middleend_tests/passes_tests/mark_fast_stages.cpp (hunk 2 above)
Comment (AndrewBakalinIntel) [reply inference-engine/tests/unit/vpu/middleend_tests/passes_tests/mark_fast_stages.cpp]: Fixed.

**Comment by AndrewBakalinIntel** [comment]
@Maxim-Doronin could you please take and merge if it's ok?

File: inference-engine/src/vpu/graph_transformer/src/middleend/passes/mark_fast_stages.cpp [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    void run(const Model& model) override {
+        VPU_PROFILE(markFastStages);
+
+        for (const auto& stage : model->getStages()) {
+            const auto& outputs = stage->outputs();
+            const auto& itSlowStage = std::find_if(outputs.begin(), outputs.end(), [](const Data& output) {
+                return output->desc().totalDimSize() > 100;
+            });
+
+            if (itSlowStage == outputs.end()) {
```
Comment (Maxim-Doronin) [inline inference-engine/src/vpu/graph_transformer/src/middleend/passes/mark_fast_stages.cpp:22]: How do you feel about such a method?
`std::all_of(..., {output->desc().totalDimSize() <= 100})`

Comment (AndrewBakalinIntel) [reply inference-engine/src/vpu/graph_transformer/src/middleend/passes/mark_fast_stages.cpp]: It looks way better now. Thanks!

**Comment by AndrewBakalinIntel** [comment]
@Maxim-Doronin precommit has passed.

**Comment by gladilov-gleb** [comment]
Don't see tests

**Pull request merged**